### ISSUE TITLE
Add support for protection level EXTERNAL_VPC keys

### ISSUE THREAD

**Issue by romanini-ciandt** [terraform-google-modules/terraform-google-kms#136]
### TL;DR

[google_kms_crypto_key](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/kms_crypto_key) resource supports 4 types of [protection_level](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/kms_crypto_key#protection_level) ("SOFTWARE", "HSM", "EXTERNAL", "EXTERNAL_VPC") while kms module [key_protection_level](https://github.com/terraform-google-modules/terraform-google-kms/blob/6531847bb1e7041d08851a06d71368402d3aeb3f/variables.tf#L105) lists only "SOFTWARE" and "HSM" as valid options.

### Terraform Resources

```markdown
google_kms_crypto_key
```

### Detailed design

```markdown
Looks like key_protection_level input would already accept "EXTERNAL_VPC", it is just missing in "possible values" example string list. The new input field that we would need to implement is crypto_key_backend in order to make "EXTERNAL_VPC" key protection level type fully work.
```
